src/Microsoft.ServiceFabric.AspNetCore/ServiceFabricMiddleware.cs

```diff
@@ -85,6 +85,16 @@ public ServiceFabricMiddleware(RequestDelegate next, string urlSuffix)
                     return;
                 }
 
+                context.Response.OnStarting(() =>
```

Comment:
Please make this pull request into develop branch.
Expose an option for this in ServiceFabricIntegrationOptions (its flag), plumb it to the middleware and modify the response to add this header when the option is specified in ServiceFabricIntegrationOptions.